scripts/codestyle/run_uncrustify.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+if [ $# -gt 3 ]; then
+	echo "Too many parameters specified"
+	exit 1
+fi
+
+VERBOSE=false
+HASHES=()
+
+while [[ $# -gt 0 ]]
+do
+	key="$1"
+	case $key in
+	    -v|--verbose)
+	    VERBOSE=true
+	    ;;
+        -h|--help)
+		echo "run_uncrustify.sh [options] [<SHA1> <SHA2>]"
+		echo ""
+        echo "If SHA is provided to the script, uncrustify checks code style of files that are different in SHA and HEAD. "
+        echo "Otherwise uncommited diff is checked."
+        echo ""
+        echo "Options:"
+        echo "-v|--verbose : shows uncrustify suggestions for code style. When not set, only file names with differences are shown."
+        echo "-h|--help    : shows this message"
+        exit 1
+        ;;
+	    *)
+	    HASHES+=($key)
+	    ;;
+	esac
+	shift
+done
+
+if [ ${#HASHES[@]} -eq 2 ];
+then 
+	diffargs="${HASHES[0]} ${HASHES[1]}"
+fi
+
+files=$(git diff --name-only $diffargs);
+#files=$(find ../../ -type f -name "*.[ch]") #all files - I'll leave it just in case
+
+folders_to_format=$(cat folders_to_format.txt)
```

Comment:
Thanks, it's better now. 

Regarding `cd`, it's is error-prone, and seems like there is an error. When I change `src/kernel/init.c`, run `./scripts/codestyle/run_uncrustify.sh` (from top directory), then `scripts/src/kernel/init.c` appear. After some debugging, seems like `$item` is `../../src/kernel/init.c` and temporary files created in `out/../../src/.....`

Also, output changed to 
```
Some files don't comply with code style:
../../src/kernel/init.c
```
I'd expect `src/kernel/init.c` in output